dts/bindings/serial/cdns,uart.yaml

```diff
@@ -0,0 +1,39 @@
+# *************************************************************************
+# Copyright Meta Platforms, Inc. and its affiliates.
+# SPDX-License-Identifier: Apache-2.0
+# *************************************************************************
+
+description: This binding gives a base representation of the CADENCE UART IP6528
+
+compatible: "cdns,uart"
+
+include: [interrupt-controller.yaml, base.yaml]
+include: uart-controller.yaml
+
+properties:
+    current-speed:
+      type: int
+      required: true
+      description: Baud Rate in bps.
+    clock-frequency:
+      type: int
+      required: true
+      description: Frequency(Hz) of Clock to the UART IP.
+    bdiv:
+      type: int
+      required: true
+      description: Baud Rate Divide register value.
+    do-not-init:
+      type: boolean
+      required: false
+      description: Tells the UART driver to skip device initialization
+    reg:
+      required: true
+    parity:
+      required: false
+      type: string
+      default: "even"
```

Comment:
@XenuIsWatching I have kept the parity default as "even" to match the IP specification. I believe, initializing it to "none" as default might cause confusion b/w spec & code.

Reply:
The default is set to none from `uart-controller.yaml` which every other zephyr uart controller follows "none" as it's default. I would say this would be more confusing from a zephyr developer perspective.